gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +207,766 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+            check_description=check_description,
+            n_fail=n_fail,
+            display_fields=field_check_details[check_description].display_fields,
+            verbose=verbose,
+            show_percent_sites=show_percent_sites,
+            ht_count=ht_count,
+        )
+
+
+def subset_freq_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    subsets: List[str],
+    verbose: bool,
+    show_percent_sites: bool = True,
+    delimiter: str = "-",
+    metric_first_label: bool = True,
+) -> None:
+    """
+    Perform sanity checks on frequency data in input Table.
+
+    Check:
+        - Number of sites where callset frequency is equal to a subset frequency (raw and adj)
+            - eg. t.info.AC-adj != t.info.AC-subset1-adj 
+        - Total number of sites where the allele count annotation is defined (raw and adj)
+        
+    :param t: Input MatrixTable or Table.
+    :param subsets: List of sample subsets.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
+    :param show_percent_sites: If true, show the percentage and count of overall sites that fail; if False, only show the number of sites that fail.
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :param metric_first_label: If True, metric precedes label group, e.g. AC-afr-male. If False, label group precedes metric, afr-male-AC.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    field_check_expr = {}
+    field_check_details = {}
+    for subset in subsets:
+        if subset != "":
+            subset += delimiter
+            for field in ["AC", "AN", "nhomalt"]:
+                for group in ["adj", "raw"]:
+                    logger.info(
+                        "Comparing subset %s frequencies to entire callset", subset
+                    )
+                    check_field_left = f"{field}{delimiter}{group}"
+                    if metric_first_label:
+                        check_field_right = f"{field}{delimiter}{subset}{group}"
+                    else:
+                        check_field_right = f"{subset}{field}{delimiter}{group}"
+                    field_check_expr, field_check_details = make_field_check_dicts(
+                        field_check_expr=field_check_expr,
+                        field_check_details=field_check_details,
+                        check_description=f"{check_field_left} != {check_field_right}",
+                        cond_expr=t.info[check_field_left] == t.info[check_field_right],
+                        display_fields=hl.struct(
+                            **{
+                                check_field_left: t.info[check_field_left],
+                                check_field_right: t.info[check_field_right],
+                            }
+                        ),
+                    )
```

Comment:
PART OF LARGE SUGGESTION
```suggestion

                    field_check_expr[f"{check_field_left} != {check_field_right}"] = {
                        "expr": hl.agg.count_where(
                            t.info[check_field_left] == t.info[check_field_right]
                        ),
                        "display_fields": hl.struct(
                            **{
                                check_field_left: t.info[check_field_left],
                                check_field_right: t.info[check_field_right],
                            }
                        ),
                    }
```